### PR TITLE
chore: migrate lumberjack core, test-utils, http-driver and console-driver

### LINKED ISSUE #1: Migrate Internal tooling to the Nx Monorepo
The internal tools are used by the core of the project and need to be migrated first.

### PR DESCRIPTION
**Pull request by NachoVazquez**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

The new repository hasn't the lumberjack core, http-driver, console-driver and internal tooling.

Fixes: #1 #4 #3 

## What is the new behavior?

The new repository has the lumberjack core, http-driver, console-driver and internal tooling.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/internal/console-driver/test-util/src/lib/noop-console/noop-console.module.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/* eslint-disable @nrwl/nx/enforce-module-boundaries */
```
Comment (LayZeeDK) [inline libs/internal/console-driver/test-util/src/lib/noop-console/noop-console.module.ts:1]: issue: Let's create an issue to get rid of circular dependencies and/or fix module boundary lint rules.

File: jest.preset.js [hunk 2]
```diff
@@ -1,3 +1,3 @@
 const nxPreset = require('@nrwl/jest/preset').default;
 
-module.exports = { ...nxPreset };
+module.exports = { ...nxPreset, resetMocks: true };
```
Comment (LayZeeDK) [inline jest.preset.js:8]: question: Why is this necessary? (Add a comment explaining why)

File: libs/internal/console-driver/test-util/src/lib/spy-console/spy-console.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    spy = new SpyConsole();
+  });
+
+  it('can be passed a lot of arguments', () => {
+    const hundredNumbers = Array(100)
+      .fill(undefined)
+      .map((_, index) => index + 1);
+
+    // eslint-disable-next-line prefer-spread
+    spy.debug.apply(spy, hundredNumbers);
```
Comment (LayZeeDK) [inline libs/internal/console-driver/test-util/src/lib/spy-console/spy-console.service.spec.ts]: suggestion: Follow the lint rules (or adjust them if feeling strongly about going against their current settings)
```suggestion
    spy.debug(...hundredNumbers);
```

File: libs/ngworker/lumberjack/src/lib/configuration/lumberjack-root.module.spec.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { LumberjackRootModule } from './lumberjack-root.module';
+
+const optionalAngularDependency = undefined;
+
+describe(LumberjackRootModule.name, () => {
+  it('guards against being registered in multiple injectors', () => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (LayZeeDK) [inline libs/ngworker/lumberjack/src/lib/configuration/lumberjack-root.module.spec.ts]: suggestion: I would love for us to use `// @ts-expect-error` annotation comments instead if possible

File: libs/ngworker/lumberjack/src/lib/configuration/lumberjack-root.module.spec.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    const rootInjectorInstance = new LumberjackRootModule(optionalAngularDependency as any);
+
+    expect(() => new LumberjackRootModule(rootInjectorInstance)).toThrowError(/multiple injectors/);
+  });
+
+  it('does not guard the first injector that registers it', () => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (LayZeeDK) [inline libs/ngworker/lumberjack/src/lib/configuration/lumberjack-root.module.spec.ts]: suggestion: I would love for us to use `// @ts-expect-error` annotation comments instead if possible

File: libs/ngworker/lumberjack/src/lib/formatting/lumberjack-format-log.spec.ts [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+import { LumberjackLog } from '../logs/lumberjack.log';
+import { LumberjackTimeService } from '../time/lumberjack-time.service';
+
+import { lumberjackFormatLog } from './lumberjack-format-log';
+
+function parseFormattedLog(formattedLog: string) {
+  // eslint-disable-next-line no-useless-escape
+  const formattedLogPattern = /^([a-z]+) ([0-9\.:\-TZ]+) (\[(.+)\] )?(.*)$/;
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [_, level, timestamp, taggedScopeWithEndingSpace = '', scope = '', message] =
```
Comment (LayZeeDK) [inline libs/ngworker/lumberjack/src/lib/formatting/lumberjack-format-log.spec.ts]: suggestion: We might remove this lint rule deactivation by leaving out the underscore assignment.
```suggestion
  const [, level, timestamp, taggedScopeWithEndingSpace = '', scope = '', message] =
```

File: libs/ngworker/lumberjack/src/lib/formatting/lumberjack-format-log.spec.ts [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+import { LumberjackLevel } from '../logs/lumberjack-level';
+import { LumberjackLogLevel } from '../logs/lumberjack-log-level';
+import { LumberjackLog } from '../logs/lumberjack.log';
+import { LumberjackTimeService } from '../time/lumberjack-time.service';
+
+import { lumberjackFormatLog } from './lumberjack-format-log';
+
+function parseFormattedLog(formattedLog: string) {
+  // eslint-disable-next-line no-useless-escape
+  const formattedLogPattern = /^([a-z]+) ([0-9\.:\-TZ]+) (\[(.+)\] )?(.*)$/;
```
Comment (LayZeeDK) [inline libs/ngworker/lumberjack/src/lib/formatting/lumberjack-format-log.spec.ts:15]: question: Which one is the useless escape (`\`)?

File: libs/ngworker/lumberjack/tsconfig.spec.json [hunk 8]
```diff
@@ -5,6 +5,9 @@
     "module": "commonjs",
     "types": ["jest", "node"]
   },
+  "typeAcquisition": {
+    "enable": true,
+    "include": ["jest"]},
```
Comment (LayZeeDK) [inline libs/ngworker/lumberjack/tsconfig.spec.json]: question: Why is this needed? According to [the TypeScript docs](https://www.typescriptlang.org/tsconfig#typeAcquisition), this is only relevant for JavaScript.

**Review by LayZeeDK — CHANGES_REQUESTED** [review]

File: libs/internal/console-driver/test-util/src/lib/noop-console/noop-console.module.ts (hunk 1 above)
Comment (NachoVazquez) [reply libs/internal/console-driver/test-util/src/lib/noop-console/noop-console.module.ts:1]: Yep, I had it as a draft, thank you!

https://github.com/ngworker/lumberjack-next/issues/6

File: libs/ngworker/lumberjack/src/lib/formatting/lumberjack-format-log.spec.ts (hunk 7 above)
Comment (NachoVazquez) [reply libs/ngworker/lumberjack/src/lib/formatting/lumberjack-format-log.spec.ts:15]: It is the one `([0-9\.:\-TZ]+)` before the dot (\.).

I'm not sure that it is really useless, so let's try it out later.

File: libs/ngworker/lumberjack/tsconfig.spec.json (hunk 8 above)
Comment (NachoVazquez) [reply libs/ngworker/lumberjack/tsconfig.spec.json]: Nothing, me trying to fix that the Jest name wasn't available. But you fixed in another commit.
